collect_app/src/main/java/org/odk/collect/android/activities/viewmodels/FormDownloadListViewModel.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2019 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.odk.collect.android.activities.viewmodels;
+
+import android.arch.lifecycle.ViewModel;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.logic.FormDetails;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+
+public class FormDownloadListViewModel extends ViewModel {
+    private HashMap<String, FormDetails> formNamesAndURLs = new HashMap<>();
+    private final HashMap<String, Boolean> formResult = new HashMap<>();
+
+    private final ArrayList<HashMap<String, String>> formList = new ArrayList<>();
+
+    private final LinkedHashSet<String> selectedForms = new LinkedHashSet<>();
+
+    private String[] formIdsToDownload;
+
+    private String alertTitle;
+    private String progressDialogMsg;
+    private String alertDialogMsg;
+    private String url;
+    private String username;
+    private String password;
+
+    private boolean progressDialogShowing;
+    private boolean alertShowing;
+    private boolean cancelDialogShowing;
+    private boolean shouldExit;
+    private boolean loadingCanceled;
+    // Variables for the external app intent call
+    private boolean isDownloadOnlyMode;
```

Comment:
Can we improve this comment? I know you just copied what was there before, but it's unclear what it actually means (and it should probably be JavaDoc).

Reply:
Fixed.